src/files/routes.rs

```diff
@@ -49,7 +63,66 @@ async fn upload_file(
         }
     }
 
-    Ok((StatusCode::CREATED, Json(FileUploaded { id })))
+    Ok((StatusCode::CREATED, Json(FileWrapper { id })))
+}
+
+/// Axum service to handle file download requests
+/// FIXME: On integration, we need to do auth via token `Session` extrating on the request
+use axum_macros::debug_handler;
+#[debug_handler]
+async fn download_file(
+    // State(app): State<Arc<AppState>>,
+    // FIXME: harden extractor to only accept "*.filetype" or similar to prevent malicious client requests for "../../something.sh" and the like
```

Comment:
The easy way to do this is to use a UUID for the filename. Then when the string is received, parse it as a UUID. If unsuccessful then return the failure/error. if successful, then convert the UUID instance back to a string for the filename.

Reply:
We will parse a `Path<String>` to a UUID in a download request, see bfa91cff50b59d20008d4cea16a605709480fe30